File: frontend/src/types/api/otp.ts
export interface ExtractedOtpCodeResponse {
  id: string;
  mail_account_id: string;
  mail_uid: string;
  sender_email: string | null;
  mail_subject: string | null;
  code: string;
  service: string | null;
  code_type: string;
  expires_at: string | null;
  is_expired: boolean;
  created_at: string;
  updated_at: string;
}

export interface ExtractedOtpCodeListResponse {
  items: ExtractedOtpCodeResponse[];
  total: number;
  page: number;
  per_page: number;
  pages: number;
}

export type ListOtpCodesApiOtpCodesGetSort = "newest" | "oldest" | "service" | "expiry";

export interface ListOtpCodesApiOtpCodesGetParams {
  page?: number;
  per_page?: number;
  service?: string;
  code_type?: string;
  active_only?: boolean;
  sort?: ListOtpCodesApiOtpCodesGetSort;
}

File: backend/app/schemas/otp.py
"""Pydantic schemas for OTP code API requests and responses."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.schemas.plugin_base import PluginResponseBase


class ExtractedOtpCodeResponse(PluginResponseBase):
    """Response schema for an extracted OTP code."""

    id: UUID
    code: str
    description: str | None = None
    service: str | None = None
    code_type: str
    url: str | None = None
    expires_at: datetime | None = None
    is_expired: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ExtractedOtpCodeListResponse(BaseModel):
    """Paginated list of extracted OTP codes."""

    items: list[ExtractedOtpCodeResponse]
    total: int
    page: int
    per_page: int
    pages: int
